pkg/query/selection.go

```diff
@@ -109,13 +111,23 @@ type Criterion struct {
 	RightOp []string
 	// Type is the type of the query
 	Type CriterionType
+	// TemplateParametersMap contains the template parameters of Criterion supporting sub-queries
+	TemplateParametersMap map[string]interface{}
 }
 
 // ByField constructs a new criterion for field querying
 func ByField(operator Operator, leftOp string, rightOp ...string) Criterion {
 	return NewCriterion(leftOp, operator, rightOp, FieldQuery)
 }
 
+func ByIdNotExist(subQuery string) Criterion {
```

Comment:
could be nice to get the template params here already, and not have to set it afterwards

Reply:
Good point! thanks!
It's done.